src/js/model/string/string.js

```diff
@@ -0,0 +1,43 @@
+var AMPParse = AMPParse || {};
+
+AMPParse.buildString = function (hexadecimal) {
+    nibblesUsed = 0; //int
+    nullTerminate = false; //bool
+    returnString = ""; //string
+    tempChars = "00";
+    
+    // Defensive programming
+    if (typeof hexadecimal === "undefined" || typeof hexadecimal !== "string") {
```

Comment:
You can probably get rid of the "undefined" test. We just have to verify that it's not a string to throw the reference error.